Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerInterface.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h>
+
+namespace AzToolsFramework
+{
+    //! The AZ::Interface of the central editor mode tracker for all viewports.
+    class ViewportEditorModeTrackerInterface
+    {
+    public:
+        AZ_RTTI(ViewportEditorModeTrackerInterface, "{7D72A4F7-2147-4ED9-A315-E456A3BE3CF6}");
+
+        virtual ~ViewportEditorModeTrackerInterface() = default;
+
+        //! Enters the specified editor mode for the specified viewport.
+        virtual void EnterMode(const ViewportEditorModeInfo& viewportEditorModeInfo, ViewportEditorMode mode) = 0;
+       
+        //! Exits the specified editor mode for the specified viewport.
+        virtual void ExitMode(const ViewportEditorModeInfo& viewportEditorModeInfo, ViewportEditorMode mode) = 0;
```

Comment:
Reposting the feedback given offline for reference.

I'd change the names of these functions to something like RegisterModeEntered() or something similar. The current name makes it feel like they're a request, which they arent.

Additionally, possible issues should be returned as an AZ::Outcome object instead of being output as warnings, so that the caller can determine the best mitigation (for example, if the caller tries to enter a mode that is already active, it would receive a negative outcome and can then fix the issue, or print the error as appropriate).

Reply:
Good ideas! :+1:

Reply:
great suggestion, i'll implement this and amend the tests accordingly 👍